### PR TITLE
JDK-8307128: Open source some drag and drop tests 4

### PR DESCRIPTION
**Pull request by alisenchung**
added DnD tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307128](https://bugs.openjdk.org/browse/JDK-8307128): Open source some drag and drop tests 4

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [1c58b757](https://git.openjdk.org/jdk/pull/13765/files/1c58b75775dfb999a38b99aa7644365f1b21f422)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13765/head:pull/13765` \
`$ git checkout pull/13765`

Update a local copy of the PR: \
`$ git checkout pull/13765` \
`$ git pull https://git.openjdk.org/jdk.git pull/13765/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13765`

View PR using the GUI difftool: \
`$ git pr show -t 13765`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13765.diff">https://git.openjdk.org/jdk/pull/13765.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13765#issuecomment-1532226954)

### CODE REVIEW

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/dnd/MozillaDnDTest.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+        MozillaDnDTest test = new MozillaDnDTest();
+        if (args.length > 0) {
+            test.run(args);
+        } else {
+            EventQueue.invokeAndWait(test::init);
+            try {
+                test.start();
+            } finally {
+                EventQueue.invokeAndWait(() -> {
+                    if(test.frame != null) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MozillaDnDTest.java:97]:
```suggestion
                    if (test.frame != null) {
```

File: test/jdk/java/awt/dnd/MozillaDnDTest.java [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+                    }
+                    while (err.available() > 0) {
+                        pres.stderr += (char)err.read();
+                    }
+                    // Ask the process for its exitValue. If the process
+                    // is not finished, an IllegalThreadStateException
+                    // is thrown. If it is finished, we fall through and
+                    // the variable finished is set to true.
+                    pres.exitValue = p.exitValue();
+                    finished  = true;
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MozillaDnDTest.java:347]:
```suggestion
                    finished = true;
```

File: test/jdk/java/awt/dnd/NestedHeavyweightDropTargetTest.java [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+        if ((dtde.getSourceActions() & DnDConstants.ACTION_COPY) != 0) {
+            dtde.acceptDrop(DnDConstants.ACTION_COPY);
+        } else {
+            dtde.rejectDrop();
+        }
+
+        DataFlavor[] dfs = dtde.getCurrentDataFlavors();
+        Component comp = null;
+
+        if(dfs != null && dfs.length >= 1) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/NestedHeavyweightDropTargetTest.java:478]:
```suggestion
        if (dfs != null && dfs.length >= 1) {
```

File: test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+            throw new RuntimeException("FAILED: Incorrect number of data flavors.");
+        }
+
+        for (int i = 0; i < dfs.length; i++) {
+
+            DataFlavor flavor = dfs[i];
+            Integer transferNumber = null;
+
+            if (flavor.getRepresentationClass().equals(Integer.class)) {
+                try{
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java:378]:
```suggestion
                try {
```

File: test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+                    transferNumber = (Integer)t.getTransferData(flavor);
+                } catch(Exception e) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java:380]:
```suggestion
                } catch (Exception e) {
```

File: test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+                    e.printStackTrace();
+                    throw new RuntimeException("FAILED: Cannot get data: " +
+                                               flavor.getMimeType());
+                }
+            }
+
+            boolean supportedFlavor = false;
+            for (int j = 0; j < TransferableNumber.NUM_DATA_FLAVORS; j++) {
+                int number = (i + j) % TransferableNumber.NUM_DATA_FLAVORS;
+                try{
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java:390]:
```suggestion
                try {
```

File: test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+                    if (flavor.equals(new IntegerDataFlavor(number))) {
+                        if (!(new Integer(number).equals(transferNumber))) {
+                            throw new RuntimeException("FAILED: Invalid data \n" +
+                                                       "\tflavor : " + flavor +
+                                                       "\tdata   : " + transferNumber);
+                        }
+                        supportedFlavor = true;
+                        break;
+                    }
+                } catch(ClassNotFoundException cannotHappen) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/MultiDataFlavorDropTest.java:400]:
```suggestion
                } catch (ClassNotFoundException cannotHappen) {
```

File: test/jdk/java/awt/dnd/NativeDragJavaDropTest.java [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+        if ((dtde.getSourceActions() & DnDConstants.ACTION_COPY) != 0) {
+            dtde.acceptDrop(DnDConstants.ACTION_COPY);
+        } else {
+            dtde.rejectDrop();
+        }
+
+        DataFlavor[] dfs = dtde.getCurrentDataFlavors();
+
+        if(dfs != null && dfs.length >= 1) {
```
Comment (turbanoff) [inline test/jdk/java/awt/dnd/NativeDragJavaDropTest.java:140]:
```suggestion
        if (dfs != null && dfs.length >= 1) {
```

File: test/jdk/java/awt/dnd/MouseExitGestureTriggerTest.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        final Robot robot = new Robot();
+        Thread.sleep(FRAME_ACTIVATION_TIMEOUT);
+
+        clickRootNode(robot);
+        clickRootNode(robot);
+        clickRootNode(robot);
+
+        Thread.sleep(RECOGNITION_TIMEOUT);
+
+        if (recognized) {
```
Comment (mrserb) [inline test/jdk/java/awt/dnd/MouseExitGestureTriggerTest.java:107]: This field is set on EDT, and used here on main w/o synchronization.

**Pull request closed without merging**